### PR TITLE
Category implementation in ModifyQuestionById

### LINKED ISSUE #14: REQ-013: Category System Implementation
# Category System Implementation

## Description
Modify the question structure to include categories and implement category-based functionality.

### Structure Update
```c
typedef struct Question {
    int id;
    int category;    // New field (1-6)
    char question[MAX_STRING];
    char options[3][MAX_STRING];
    int correct_answer;
    struct Question* next;
} Question;
```

### Categories (1-6)
1. Mexican History
2. General History
3. Geography
4. Sports, Movies, Books, Art
5. Science and Biology
6. Spanish, Philosophy and Religion

### Required Features
* Update question structure with category field
* Create function to filter questions by category
* Add category display in gameplay
* Implement category selection menu
* Update modifyQuestionById to handle category updates

### Integration Requirements
1. Question Structure Changes:
   - ID system integration
   - Category system integration
   - File persistence adaptation

2. Game Logic Updates:
   - Category display during gameplay
   - Category selection integration
   - Question filtering by category

3. File Operations:
   - Updated file format with new fields
   - Data validation for new fields
   - Backwards compatibility

### Required Functions
```c
void displayQuestionsByCategory(Question* head, int category);
```

### PR DESCRIPTION
**Pull request by leonardomedinaa**
**Description**
This PR updates the modifyQuestionById function to allow modifications to the category field in questions. The changes include:

- Support for category editing: Added functionality to update the category of an existing question.
- Category validation: Ensures that the entered category is within the allowed range (1-6).
- Improved interface: Displays the current category of the question and prompts the user to update it during the modification process.

**Testing**

- Verified successful updates for questions with various categories.
- Ensured proper validation for categories within the valid range.
- Handled edge cases where invalid categories (outside the range) were entered.

Closes #14

### CODE REVIEW

File: marathon_quiz.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ int main(){
             case 6: 
 			    printf("\nEnter the id of the question to modify: ");
             	scanf("%d",&questionId);
-				modifyQuestionById(questionHead, questionId);
+				modifyQuestionById(questionHead, questionId, category);
```
Comment (gfdfranco) [inline marathon_quiz.cpp]: Please fix the indentation of this code block

**Review by gfdfranco — CHANGES_REQUESTED** [review]
Just a little fix, ty !

**Review by gfdfranco — APPROVED** [review]
LGTM

**Review by Friedamariao — APPROVED** [review]
LGTM

**Review by OliverHerrera140 — APPROVED** [review]
LGTM

**Pull request merged**